assembly/src/parsers/ast/mod.rs

```diff
@@ -78,39 +79,40 @@ impl ProgramAst {
 /// A module can only have exported and local procedures, but no body.
 #[derive(Debug, Eq, PartialEq)]
 pub struct ModuleAst {
-    pub procedures: ProcMap,
+    pub procedures: LocalProcMap,
 }
 
 impl ModuleAst {
     /// Returns byte representation of the `ModuleAst.
     pub fn to_bytes(&self) -> Vec<u8> {
-        let mut module_ast_bytes = ByteWriter::new();
+        let mut byte_writer = ByteWriter::new();
 
         // procedures
-        module_ast_bytes.write_u16(self.procedures.len() as u16);
-        for (key, value) in self.procedures.iter() {
-            module_ast_bytes
-                .write_string(key)
+        byte_writer.write_u16(self.procedures.len() as u16);
+
+        for (_, proc) in sort_procs_by_index(&self.procedures) {
+            byte_writer
+                .write_string(&proc.name)
```

Comment:
Do we need to write procedure name here? I think we write it as a part of `proc.write_into()` statement below.